pkg/miniogw/gateway-storj.go

```diff
@@ -224,18 +224,18 @@ func (s *storjObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
 	}
 	if len(items) > 0 {
 		for _, item := range items {
-			p := item.Path
-			if recursive {
-				p = path.Join(prefix, p)
+			path := item.Path
+			if recursive && prefix != "" {
+				path = storj.JoinPaths(strings.TrimSuffix(prefix, "/"), path)
```

Comment:
`prefix + path` should work directly here otherwise it means somewhere the splitting is problematic.

Reply:
Yes, you are right, but currently we need this as a workaround. Imagine we have two files:
- a
- a/b

Listing recursively with prefix `a/` will correctly return `a/b`.
But listing recursively with prefix `a` will return the incorrect `ab` (the slash in-between is missing).

So we need this workaround for now. Perhaps, we can do just `prefix + path` if we look into the PointerDB iterator to fix the issue, but let's do it in another PR.